numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -630,7 +630,7 @@ static NPY_INLINE npy_longdouble
 hpy_string_to_long_double(HPyContext *ctx, HPy op)
 {
     const char *s;
-    const char *end;
+    char *end;
```

Comment:
this caused many warnings, so either my change or all the uses of `end` must be `const char`